src/components/PopoverButton/Popover.js

```diff
@@ -0,0 +1,117 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Manager, Reference, Popper } from 'react-popper';
+import styled from 'react-emotion';
+
+const modifiers = {
+  offset: { enabled: true, offset: '0,10' },
+  flip: { enabled: true }
+};
+
+class Popover extends Component {
+  static propTypes = {
+    /**
+     * function rendering the popover
+     */
+    renderPopover: PropTypes.func.isRequired,
+    /**
+     * function rendering the reference (button or something clickable)
+     */
+    renderReference: PropTypes.func.isRequired,
+    /**
+     * placement of the popover relative to the button
+     */
+    placement: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
+    /**
+     * alignment of the popover relative to the button
+     */
+    align: PropTypes.oneOf(['start', 'end', 'center'])
+  };
+
+  static defaultProps = {
+    placement: 'bottom',
+    align: 'start'
+  };
+
+  state = { isOpen: false };
+
+  componentDidMount() {
+    document.addEventListener('click', this.onDocumentClick, false);
```

Comment:
I was thinking about it, but nothing came to mind. Also, giving control to the user over how to close the popover would complicate the component big time and I am not sure it can be done without some heavy configuration options.

As discussed, the only thing that I think we can do is to have the option for the calendar to show on mouseover/mouseleave which will bring the component one level low and capable to facilitate building tooltip abstraction over it.

Anyway, I don't think we should go for it, immediately. Let's see how this works and what are the tradeoffs, first.